### DISCUSSION TITLE
Error: fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to establish connection to NetBox API"}

### DISCUSSION THREAD

**Discussion by digital-fortress** [netbox-community/netbox#15674] [Q&A] [answered]
Let me first give a statement - I am not an expert in ansible. Just doing this part of job.

Now I am reaching here after exhausting all troubleshooting options.

While running my fist ansible playbook for netbox, I get this error and I am unable to move forward. I have verified my API is working good.

`more 1.yml
- hosts: localhost
  vars:
    validate_certs: false
    netbox_url: "my_netbox"
    netbox_token: "api_token"
  tasks:
    - name: Check HTTPS access to Netbox
      uri:
        url: "{{ netbox_url }}"
        method: GET
        return_content: yes
        status_code: 200
        validate_certs: "{{ validate_certs }}"
      register: netbox_check

    - name: Proceed if HTTPS access is working
      block:
        - name: "Example for state: present"
          netbox.netbox.netbox_device:
            netbox_url: "{{ netbox_url }}"
            netbox_token: "{{ netbox_token }}"
            data:
              name: "Test Device"
              device_type: "C9410R"
              device_role: "DC Device (IOC Managed)"
              site: "Bangalore - RGA Tech Park"
            state: present
      when: netbox_check.status == 200
	  

Getting the below error.

ansible-playbook 1.yml
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Check HTTPS access to Netbox] *****************************************************************************************************************************************************************************
ok: [localhost]

TASK [Example for state: present] *******************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to establish connection to NetBox API"}

PLAY RECAP ******************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

Can someone help ?

**Comment by digital-fortress** [comment]
Can an experienced person , help me with this ?

**Reply by candlerb** [reply, marked as answer]
1. Please edit your original post and delimit your quoted text with three backticks (` ``` `) before and after, on a line of their own. Then it will be readable.
2. This question is missing essential background information (e.g. version of Netbox, version on Ansible and of the module you installed)
3. In general, questions about Ansible (or a third party Ansible plugin, in this case [netbox.netbox.netbox_device](https://docs.ansible.com/ansible/latest/collections/netbox/netbox/netbox_device_module.html)) would IMO be best asked on an Ansible list.
4. If Ansible says it cannot establish connection to Netbox, then it's probably correct. You should look at logs on your server side to see what's going on.

One possibility is that the API token you're providing is invalid. Another possibility is that you don't have a valid certificate. I notice you set `validate_certs` on your first task, but not on the second.

But essentially this looks like something you need to debug on your system; there is no magic wand we can wave for you.

**Comment by digital-fortress** [comment]
Fixed with this 

# Disable SSL warnings (optional, if you have SSL warnings)
requests.packages.urllib3.disable_warnings(requests.packages.urllib3.exceptions.InsecureRequestWarning)
